### PR TITLE
MULE-17450: Http server lookup does not support domains

### PR DESCRIPTION
**Pull request by lucianoRM**

### CODE REVIEW

File: src/main/java/org/mule/service/http/impl/service/HttpServiceImplementation.java [hunk 1]
```diff
@@ -70,8 +77,17 @@ public HttpServerFactory getServerFactory() {
...
+    Optional<String> appName = registry.lookupByName(APP_NAME_PROPERTY);
+    Optional<String> domainName = registry.lookupByName(DOMAIN_NAME_PROPERTY);
+    if (domainName.isPresent() && appName.isPresent()) {
+      return new ContextHttpServerFactoryAdapter(appName.get(), domainName.get(), listenerConnectionManager);
+    }
+    return domainName
+        .map(d -> new ContextHttpServerFactoryAdapter(d, listenerConnectionManager))
+        .orElseGet(() -> appName.map(a -> new ContextHttpServerFactoryAdapter(a, listenerConnectionManager))
+            .orElseThrow(() -> new MuleRuntimeException(createStaticMessage("Could not create server factory because neither "
+                + APP_NAME_PROPERTY + " nor " + DOMAIN_NAME_PROPERTY + " is defined"))));
```
Comment (afelisatti) [inline src/main/java/org/mule/service/http/impl/service/HttpServiceImplementation.java]: `are defined`

**Comment by afelisatti** [comment]
Use a MULE issue for this.

File: src/main/java/org/mule/service/http/impl/service/HttpServiceImplementation.java (hunk 1 above)
Comment (lucianoRM) [reply src/main/java/org/mule/service/http/impl/service/HttpServiceImplementation.java]: I also had doubts with this, but i think this is the correct form.
https://english.stackexchange.com/questions/1144/neither-michael-nor-albert-is-correct-or-neither-michael-nor-albert-are-corre

File: src/main/java/org/mule/service/http/impl/service/server/ContextHttpServerFactoryAdapter.java [hunk 2]
```diff
@@ -19,11 +19,19 @@
...
   private final String context;
   private final ContextHttpServerFactory delegate;
 
   public ContextHttpServerFactoryAdapter(String context, ContextHttpServerFactory delegate) {
     this.context = context;
+    this.parentContext = null;
+    this.delegate = delegate;
+  }
+
+  public ContextHttpServerFactoryAdapter(String context, String parentContext, ContextHttpServerFactory delegate) {
```
Comment (elrodro83) [inline src/main/java/org/mule/service/http/impl/service/server/ContextHttpServerFactoryAdapter.java]: make `parentContext` an `Optional` and remove/deprecate the previous constructor

File: src/test/java/org/mule/service/http/impl/service/HttpServiceImplementationServerFactoryTestCase.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+import org.mule.tck.SimpleUnitTestSupportSchedulerService;
+
+import java.util.Optional;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class HttpServiceImplementationServerFactoryTestCase {
```
Comment (afelisatti) [inline src/test/java/org/mule/service/http/impl/service/HttpServiceImplementationServerFactoryTestCase.java]: I believe there's a particular hierarchy that enables running this too on the EE service (pending changes there).

File: src/test/java/org/mule/service/http/impl/service/HttpServiceImplementationServerFactoryTestCase.java [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+  @Test
+  public void failureWhenServerForUnknownArtifactType() {
+    expectedException.expect(MuleRuntimeException.class);
+    expectedException.expectMessage("Unable to create HttpServerFactory for artifact with type: PLUGIN, with properties");
+    newServerFactory(empty(), empty(), PLUGIN);
+  }
+
+  @Test
+  public void failureWhenServerForAPPWithNoName() {
```
Comment (afelisatti) [inline src/test/java/org/mule/service/http/impl/service/HttpServiceImplementationServerFactoryTestCase.java]: lower case these test names to follow convention

File: src/test/java/org/mule/service/http/impl/service/HttpServiceImplementationServerFactoryTestCase.java [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    expectedException.expectMessage("Could not create server factory for policy");
+    newServerFactory(empty(), empty(), POLICY);
+  }
+
+  @Test
+  public void sameAppAccessSameServer() throws Exception {
+    final String app = "app";
+    HttpServerFactory factory = newAppServerFactory(app, empty());
+
+    //No failure means OK. Can't compare servers because they are different implementations
```
Comment (afelisatti) [inline src/test/java/org/mule/service/http/impl/service/HttpServiceImplementationServerFactoryTestCase.java]: You could assert that the results are not null at least

**Pull request merged**